File: frontend/src/types/nextGen.ts
/**
 * Type definitions for next-generation features
 * Defines interfaces for AI interactions, voice commands, AR, cloud saves, and predictions
 */
export interface SocialInteractionResponse {
  summary: string;
  suggested_follow_up: string;
  timestamp: string;
}

export interface VoiceCommandResponse {
  intent: string;
  confidence: number;
  action?: string | null;
  feedback: string;
}

export interface ARSessionResponse {
  session_id: string;
  anchor_description: string;
  instructions: string[];
  pet_data?: {
    id: string;
    name: string;
    species: string;
    breed: string;
    color_pattern: string;
    mood: string;
    stats: {
      hunger: number;
      happiness: number;
      cleanliness: number;
      energy: number;
      health: number;
    };
  } | null;
}

export interface CloudSaveResponse {
  saved_at: string;
  checksum: string;
}

export interface WeatherReactionResponse {
  condition: string;
  temperature_c: number;
  reaction: string;
  recommendation: string;
}

export interface HabitPredictionResponse {
  preferred_actions: string[];
  next_best_time: string;
  confidence: number;
  ai_suggestions?: string[];
  notification_message?: string | null;
}

export interface SeasonalEventResponse {
  event_name: string;
  message: string;
  rewards: string[];
}

File: app/schemas/next_gen.py
"""
Schemas for next-generation features (social, voice, AR, weather, habit prediction).
"""

from __future__ import annotations

from datetime import datetime
from typing import List, Optional
from uuid import UUID

from pydantic import BaseModel, Field


class SocialInteractionRequest(BaseModel):
    pet_id: UUID
    target_pet_id: UUID
    prompt: str


class SocialInteractionResponse(BaseModel):
    summary: str
    suggested_follow_up: str
    timestamp: datetime


class VoiceCommandRequest(BaseModel):
    transcript: str
    locale: str = Field(default="en-US")


class VoiceCommandResponse(BaseModel):
    intent: str
    confidence: float
    action: Optional[str]
    feedback: str


class ARSessionResponse(BaseModel):
    session_id: str
    anchor_description: str
    instructions: List[str]
    pet_data: Optional[dict] = Field(default=None, description="Pet data for AR rendering")


class CloudSavePayload(BaseModel):
    state: dict


class CloudSaveResponse(BaseModel):
    saved_at: datetime
    checksum: str


class WeatherReactionResponse(BaseModel):
    condition: str
    temperature_c: float
    reaction: str
    recommendation: str


class HabitPredictionResponse(BaseModel):
    preferred_actions: List[str]
    next_best_time: str
    confidence: float
    ai_suggestions: List[str] = Field(default_factory=list, description="AI-generated care suggestions")
    notification_message: Optional[str] = Field(default=None, description="Notification to display to user")


class SeasonalEventResponse(BaseModel):
    event_name: str
    message: str
    rewards: List[str]

